server/image_pull.go

```diff
@@ -17,7 +21,29 @@ func (s *Server) PullImage(ctx context.Context, req *pb.PullImageRequest) (*pb.P
 	if img != nil {
 		image = img.Image
 	}
-	options := &copy.Options{}
+	var (
+		username string
+		password string
+	)
+	if req.GetAuth() != nil {
+		username = req.GetAuth().Username
+		password = req.GetAuth().Password
+		if req.GetAuth().Auth != "" {
+			var err error
+			username, password, err = decodeDockerAuth(req.GetAuth().Auth)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	options := &copy.Options{
+		SourceCtx: &types.SystemContext{
+			DockerAuthConfig: &types.DockerAuthConfig{
+				Username: username,
+				Password: password,
+			},
```

Comment:
not actually, I re-pushed a fixed version I guess. If those are empty, we'll simply not read the `.docker/config.json`. The latest revision makes sure not to send anything if password is empty.